plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2721,18 +2720,27 @@ private static void validateBucketColumns(ConnectorTableMetadata tableMetadata)
         }
         Set<String> allColumns = tableMetadata.getColumns().stream()
                 .map(ColumnMetadata::getName)
-                .collect(toSet());
+                .collect(toImmutableSet());
 
         List<String> bucketedBy = bucketProperty.get().getBucketedBy();
         if (!allColumns.containsAll(bucketedBy)) {
-            throw new TrinoException(INVALID_TABLE_PROPERTY, format("Bucketing columns %s not present in schema", Sets.difference(ImmutableSet.copyOf(bucketedBy), ImmutableSet.copyOf(allColumns))));
+            throw new TrinoException(INVALID_TABLE_PROPERTY, format("Bucketing columns %s not present in schema", Sets.difference(ImmutableSet.copyOf(bucketedBy), allColumns)));
+        }
+
+        Set<String> partitionColumns = ImmutableSet.copyOf(getPartitionedBy(tableMetadata.getProperties()));
+        if (bucketedBy.stream().anyMatch(partitionColumns::contains)) {
+            throw new TrinoException(INVALID_TABLE_PROPERTY, format("Bucketing columns %s are also used as partitioning columns", Sets.intersection(ImmutableSet.copyOf(bucketedBy), partitionColumns)));
         }
 
         List<String> sortedBy = bucketProperty.get().getSortedBy().stream()
                 .map(SortingColumn::getColumnName)
                 .collect(toImmutableList());
         if (!allColumns.containsAll(sortedBy)) {
-            throw new TrinoException(INVALID_TABLE_PROPERTY, format("Sorting columns %s not present in schema", Sets.difference(ImmutableSet.copyOf(sortedBy), ImmutableSet.copyOf(allColumns))));
+            throw new TrinoException(INVALID_TABLE_PROPERTY, format("Sorting columns %s not present in schema", Sets.difference(ImmutableSet.copyOf(sortedBy), allColumns)));
+        }
+
+        if (sortedBy.stream().anyMatch(partitionColumns::contains)) {
+            throw new TrinoException(INVALID_TABLE_PROPERTY, format("Sorting columns %s are also used as partitioning columns", Sets.intersection(ImmutableSet.copyOf(sortedBy), partitionColumns)));
```

Comment:
is this illegal too?

Reply:
In Hive syntax partitioning columns need to be mentioned separately from the columns in CREATE TABLE scope and there can't be any overlap. E.g.;
```
CREATE TABLE `store_sales_test`(
    >   `ss_sold_date_sk` bigint,
    >   `ss_sold_time_sk` bigint,
    >   `ss_item_sk` bigint)
    > PARTITIONED BY (ss_item_sk bigint)
    > CLUSTERED BY (
    >   ss_item_sk)
    > INTO 4 BUCKETS;
FAILED: SemanticException [Error 10035]: Column repeated in partitioning columns
```
Also, the columns in SORTED/CLUSTERED must come from CREATE TABLE scope
```
hive> CREATE TABLE `store_sales_test`(
    >   `ss_sold_date_sk` bigint,
    >   `ss_sold_time_sk` bigint)
    > PARTITIONED BY (ss_item_sk bigint)
    > CLUSTERED BY (
    >   ss_item_sk)
    > INTO 4 BUCKETS;
```
```
hive> CREATE TABLE `store_sales_test`(
    >   `ss_sold_date_sk` bigint,
    >   `ss_sold_time_sk` bigint)
    > PARTITIONED BY (ss_item_sk bigint)
    > CLUSTERED BY (ss_sold_date_sk)
    > SORTED BY (ss_item_sk)
    > INTO 4 BUCKETS;
FAILED: SemanticException [Error 10002]: Invalid column reference  'ss_item_sk'
```